src/wp-includes/script-loader.php

```diff
@@ -2426,13 +2426,39 @@ function wp_enqueue_registered_block_scripts_and_styles() {
 function enqueue_block_styles_assets() {
 	$block_styles = WP_Block_Styles_Registry::get_instance()->get_all_registered();
 
-	foreach ( $block_styles as $styles ) {
+	foreach ( $block_styles as $block_name => $styles ) {
 		foreach ( $styles as $style_properties ) {
 			if ( isset( $style_properties['style_handle'] ) ) {
-				wp_enqueue_style( $style_properties['style_handle'] );
+
+				// If the site loads separate styles per-block, enqueue the stylesheet on render.
+				if ( wp_should_load_separate_core_block_assets() ) {
+					add_filter(
```

Comment:
In most of the places in WP core, the string is passed as the 2nd argument so plugins could have a simple way to remove it. Do you think, it should be available as an option here as well? There are also some places where a Callback is passed as the 2nd argument so I'm not sure.

Reply:
I was thinking about that when adding the callback... The reason I went with this is that plugins can still just run `wp_dequeue_style` to remove the styles. Since they can remove the styles, removing the hook itself seemed inconsequential...

Reply:
Good thinking. I think it makes my comment noop :)